packages/ddp-server/livedata_server.js

```diff
@@ -809,24 +809,33 @@ _.extend(Session.prototype, {
     self.collectionViews = {};
     self.userId = userId;
 
-    // Save the old named subs, and reset to having no subscriptions.
-    var oldNamedSubs = self._namedSubs;
-    self._namedSubs = {};
-    self._universalSubs = [];
+    // _setUserId is normally called from a Meteor method with
+    // DDP._CurrentInvocation set. But DDP._CurrentInvocation is not expected
+    // to be set inside a publish function, so we temporary unset it.
+    // Otherwise, this can lead to a problem that if a publish function is
```

Comment:
Somewhere along the line I guess I merged #8031 and pushed to this PR which I didn't want to do. I'm thinking I'll rebase this branch of your PR and update the comments and tests to work with my changes. I don't really see these changes making sense without the changes in your PR.